### PR TITLE
[Merged by Bors] - chore: split Algebra.Algebra.Basic

### PR DESCRIPTION
**Pull request by kim-em**
Splits `Algebra.Algebra.Defs` off `Algebra.Algebra.Basic`. Most imports only need the Defs file, which has significantly smaller imports. The remaining `Algebra.Algebra.Basic` is now a grab-bag of unrelated results, and should probably be split further or rehomed.

This is mostly motivated by the wasted effort during minimization upon encountering Algebra.Algebra.Basic.

### CODE REVIEW

**Review by grunweg — COMMENTED** [review]
I checked locally: this is faithful code motion. Others shall comment if the split makes sense. (The idea seems good to me, I have not thought about the details.)

**Review by Ruben-VandeVelde — APPROVED** [review]
maintainer merge

File: Mathlib/Algebra/Algebra/Basic.lean [hunk 1]
```diff
@@ -3,467 +3,33 @@ Copyright (c) 2018 Kenny Lau. All rights reserved.
...
-  commutes' := h
-  smul_def' _ _ := rfl
-  toRingHom := i
-#align ring_hom.to_algebra' RingHom.toAlgebra'
-
-/-- Creating an algebra from a morphism to a commutative semiring. -/
-def RingHom.toAlgebra {R S} [CommSemiring R] [CommSemiring S] (i : R →+* S) : Algebra R S :=
-  i.toAlgebra' fun _ => mul_comm _
-#align ring_hom.to_algebra RingHom.toAlgebra
+#align_import algebra.algebra.basic from "leanprover-community/mathlib"@"36b8aa61ea7c05727161f96a0532897bd72aedab"
```
Comment (Vierkantor) [inline Mathlib/Algebra/Algebra/Basic.lean:20]: We put `#align_import` before module docs, right?

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Review by Vierkantor — APPROVED** [review]
The following results and their corollaries could be moved to `Algebra.Defs`:

 * `mul_sub_algebraMap_commutes`
 * `map_algebraMap_mul`

But I agree they're of limited enough use that they don't belong in a `Defs` file.

The next splitting step I would suggest is the instances for `Algebra ℕ _`, `Algebra ℤ _` and `Algebra ℚ _`.

bors d+

**Comment by Vierkantor** [comment]
Oops, looks like Johan was too fast but I still want to make my comment :)

bors r-

**Comment by semorrison** [comment]
bors merge

**Pull request closed without merging**